app/services/transaction_creator.rb

```diff
@@ -6,36 +6,70 @@ class TransactionCreator
   #   transaction_creator.account
   # end
 
-  attr_reader :params, :user, :account
+  attr_reader :params, :user, :account, :limit
 
   def initialize(params, user)
     @params = params
     @user = user
   end
 
+  def check_creds
+    expired = account_to.valid_thru
+    return false if @params[:day] != expired.strftime('%d')
+    return false if @params[:month] != expired.strftime('%m')
+    true
```

Comment:
rewrite in one line `@params[:day] == expired.strftime('%d') || @params[:month] == expired.strftime('%m')` or similar, not sure with logics here. Probably better use timestamps and compare them